jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -97,40 +174,39 @@ def _expected_subset_layer_default(self, layer_state):
 
     def data(self, cls=None):
         # Grab the user's chosen statistic for collapsing data
-        if hasattr(self.state, 'function'):
-            statistic = self.state.function
-        else:
-            statistic = None
-
+        statistic = getattr(self.state, 'function', None)
         data = []
 
         for layer_state in self.state.layers:
             if hasattr(layer_state, 'layer'):
+                lyr = layer_state.layer
 
                 # For raw data, just include the data itself
-                if isinstance(layer_state.layer, BaseData):
+                if isinstance(lyr, BaseData):
                     _class = cls or self.default_class
 
                     if _class is not None:
-                        # If spectrum, collapse via the defined statistic
-                        if _class == Spectrum1D:
-                            layer_data = layer_state.layer.get_object(cls=_class,
-                                                                      statistic=statistic)
+                        cache_key = (lyr.label, statistic)
+                        if cache_key in self.jdaviz_app._get_object_cache:
+                            layer_data = self.jdaviz_app._get_object_cache[cache_key]
                         else:
-                            layer_data = layer_state.layer.get_object(cls=_class)
+                            # If spectrum, collapse via the defined statistic
+                            if _class == Spectrum1D:
+                                layer_data = lyr.get_object(cls=_class, statistic=statistic)
+                            else:
+                                layer_data = lyr.get_object(cls=_class)
```

Comment:
Will this logic ignore any masks associated with the spectrum, which could be accessible via `get_subset_object` rather than `get_object`?

Reply:
I am not familiar with how mask is supposed to work in Jdaviz. What is the workflow I should try to answer this question?

Reply:
Actually, now that I look at the diff again, these calls also predate the PR, so addressing this comment is out of scope here.